src/docdb/tree/DocDBDocumentTreeItem.ts

```diff
@@ -103,8 +112,8 @@ export class DocDBDocumentTreeItem implements IAzureTreeItem {
         let value;
         for (let field of fields) {
             value = value ? value[field] : this.document[field];
-            if (!value) {
-                break;
+            if (!value) { //Partition Key exists, but this document doesn't have a value
+                return {};
```

Comment:
Not a fan of writing a comment here. Was wondering if it makes more sense to have the following constants: 
```
fixedSizeCollectionPartitionKeyValue = undefined;
UnlimitedCollectionNoPartitionKeyValue= {};
```

The downside being it's only used in one place (here).

Reply:
FWIW, don't confuse the idea of fixed vs unlimited with the idea of having a partition key - they might be the same right now in Azure, but not necessarily always.

Reply:
Have you thought about whether a partition key with no value is the same as a partition key with {} as the value?

Reply:
Yes. Here's what I tested: 
1.Create a doc in the portal, with the partition key value of {}, no quotes. 
2. Open it in the extension. Doc opens fine, empty object has no quotes. 
3. Add another field and update. Open another document and open this document again. Works fine. 
4. Remove the partition key, update. Open another document and open this document again. Works fine. 
5. Add back the empty object partition key value. Update. Open another document and open this document again.  Works fine. 

My hypothesis is that at the server side, documents with either value ({} or no value) are placed in the same partition, and hence it has had no problem locating the document.  But the API expects a value, and the 'no partition key' value happens to be {}. 


Hope this answers the question.

Reply:
Not fully happy with setting the partition key value to {} when it has no value.  It feels like a hack (although I realize it's Azure that's enforcing it).  Personally it seems like returning undefined makes more sense, and then converting on the update/delete side.  Otherwise you're fixing up a problem at update/delete time by making a chance in code farther away.

But not sure what other effects either of these choices causes - is partitionKey used in other places where the undefined vs {} might be an issue?

If you do go with {} here, then I agree having a constant for it would be a good idea.